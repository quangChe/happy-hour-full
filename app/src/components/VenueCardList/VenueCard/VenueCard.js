import React from 'react';
import './VenueCard.scss';

export default class VenueCard extends React.Component {
  
  render() {
    if (!this.props) return null;
    
    else {
      const { handleClick, venue, width } = this.props;
      const cardStyle = {
        width,
        backgroundImage: `url(${venue.image_url})`
      };
  
      return (
        <div className="venue-card"
            style={cardStyle} >
          <div className="card-overlay">
            <p onClick={() => handleClick(venue.id)} className="name">{venue.name}</p>
          </div>
        </div>
      )
    }
  }
}