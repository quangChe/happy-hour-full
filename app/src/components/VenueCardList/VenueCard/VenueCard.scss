.venue-card {
  font-size: 3rem;
  height: 100%;
  display: inline-block;
  background-attachment: fixed;
  background-size: cover;
  scroll-snap-align: start;
  position: relative;
}

.card-overlay {
  height: 100%;
  width: 100%;
  background: linear-gradient(to left, rgba(51, 51, 51, 0.7), rgba(51, 51, 51, 0.8));

  .name {
    font-weight: 700;
    transition: 1s;
    width: 95%;
    box-sizing: border-box;
    white-space: pre-wrap;
    word-wrap: break-word;
    text-align: left;
    padding: 10px 20px 10px 30px;
    color: #dddddd;
    position: absolute;
    bottom: 10%;
    z-index: 2;
    text-shadow: 0 0 5px rgba(0, 0, 0, 0.5);
    background-clip: text, border-box; 
    
    &:hover { cursor: pointer; }
  }
  
}

@media only screen and (min-width: 400px) {
  .venue-card {
    font-size: 5rem;

    .name {
      bottom: 20vh;
      width: 100%;
      text-align: center;
    }
  }
}