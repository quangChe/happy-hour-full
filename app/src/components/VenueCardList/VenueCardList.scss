.venues-container {
  height: 100%;
  font-family: 'Helvetica';
  color: #f2f2d9;
  overflow-y: hidden;
  position: relative;

  .venues-scrollbox {
    height: 100%;
    overflow-x: auto;
    overflow-y: hidden;
    white-space: nowrap;
    scroll-snap-type: x mandatory;
    scroll-behavior: smooth;

    .status {
      position: absolute;
      font-size: 1rem;
      font-weight: 400;
      top: 20px;
      left: 50px;
      z-index: 2;
    }
  }
}


