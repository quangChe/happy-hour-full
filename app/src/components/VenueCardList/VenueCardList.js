import React from 'react';
import './VenueCardList.scss';
import VenueCard from './VenueCard/VenueCard';

export default class VenueCardList extends React.Component {
  state = {
    width: window.innerWidth,
    scrollPosition: 0,
  }

  updateDimensions = () => {
    const carousel = this.refs[this.props.id];
    this.setState({ 
      width: window.innerWidth, 
    });
    return carousel.scrollTo(0, 0);
  }

  autoScroll = () => {
    const carousel = this.refs[this.props.id];
    const { width, scrollPosition} = this.state;
    if (scrollPosition < carousel.scrollWidth) {
      this.setState({scrollPosition: scrollPosition + width})
      return carousel.scrollBy(width/3, 0);  
    }
    this.setState({scrollPosition: 0});
    return carousel.scrollTo(0, 0);
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateDimensions);
    this.AUTO_SCROLLING = window.setInterval(this.autoScroll, 8000);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateDimensions);
    clearInterval(this.AUTO_SCROLLING);
  }

  render() {
    const { id, venues, venueClick } = this.props;
    const { width } = this.state; 

    return (
      <div className="venues-container">
        <div ref={id} className="venues-scrollbox">
          { venues && venues.map(venue =>
              <VenueCard handleClick={venueClick} key={venue.id} venue={venue} width={width}/>
            )   
          }
        </div>
      </div>
    )
  }

}
