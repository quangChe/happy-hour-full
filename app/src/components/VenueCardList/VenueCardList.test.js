import React from 'react';
import { shallow, mount } from 'enzyme';

import VenueCardList from './VenueCardList';
import VenueCard from './VenueCard/VenueCard';

describe('<VenueCardList />', () => {
  const props = {
    id: '12345',
    venues: [
      {id: 1, name: 'Test 1', image_url: 'https://via.placeholder.com/150'},
      {id: 2, name: 'Test 2', image_url: 'https://via.placeholder.com/150'},
      {id: 3, name: 'Test 3', image_url: 'https://via.placeholder.com/150'}
    ],
    venueClick: () => console.log('Hello World!')
  };
  const wrapper = shallow(<VenueCardList {...props} />);
  
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should render a VenueCard for each item in venues prop', () => {
    expect(wrapper.find(VenueCard)).toHaveLength(props.venues.length);
  });
})