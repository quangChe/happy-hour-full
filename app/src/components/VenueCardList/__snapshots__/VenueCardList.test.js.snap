// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VenueCardList /> renders without crashing 1`] = `
<div
  className="venues-container"
>
  <div
    className="venues-scrollbox"
  >
    <VenueCard
      handleClick={[Function]}
      key="1"
      venue={
        Object {
          "id": 1,
          "image_url": "https://via.placeholder.com/150",
          "name": "Test 1",
        }
      }
      width={1024}
    />
    <VenueCard
      handleClick={[Function]}
      key="2"
      venue={
        Object {
          "id": 2,
          "image_url": "https://via.placeholder.com/150",
          "name": "Test 2",
        }
      }
      width={1024}
    />
    <VenueCard
      handleClick={[Function]}
      key="3"
      venue={
        Object {
          "id": 3,
          "image_url": "https://via.placeholder.com/150",
          "name": "Test 3",
        }
      }
      width={1024}
    />
  </div>
</div>
`;
